### PR TITLE
feat(secrets/k8s): upstream secrets changes, sanitization, file cache

### PR DESCRIPTION
**Pull request by ncknt**
- Proper registration of the Kubernetes secret engine

### CODE REVIEW

File: pkg/halyard/halyard.go [hunk 1]
```diff
@@ -106,6 +91,14 @@ func (s *Service) buildGenManifestsRequest(ctx context.Context, spinConfig *v1al
 	return req, nil
 }
 
+func (s *Service) addObjectToRequest(ctx context.Context, writer *multipart.Writer, param string, object interface{}) error {
```
Comment (german-muzquiz) [inline pkg/halyard/halyard.go]: Looks like `ctx` parameter is not used

File: pkg/secrets/kubernetes.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+func (k *KubernetesDecrypter) parse(params string) error {
+	tokens := strings.Split(params, "!")
+	for _, element := range tokens {
+		kv := strings.Split(element, ":")
+		if len(kv) == 2 {
+			switch kv[0] {
+			case "n":
+				k.name = kv[1]
```
Comment (german-muzquiz) [inline pkg/secrets/kubernetes.go]: Would it be useful to include secret namespace as an optional param (defaulting to what you have now)?

File: pkg/secrets/secrets.go [hunk 3]
```diff
@@ -2,90 +2,68 @@ package secrets
...
 	"os"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"strings"
 )
 
-// Decode decodes a potential value into a secret
-func Decode(ctx context.Context, val string) (string, error) {
-	return decode(decryptFunc, ctx, val)
+func init() {
+	secrets.Engines["kubernetes"] = NewSecretDecrypter
```
Comment (german-muzquiz) [inline pkg/secrets/secrets.go]: Function name could be changed to something indicating it's a kubernetes specific decrypter, `NewKubeSecretDecrypter`?

File: pkg/secrets/kubernetes.go (hunk 2 above)
Comment (ncknt) [reply pkg/secrets/kubernetes.go]: It's by design, we don't want to mount secrets from a different namespace.

File: pkg/secrets/secrets.go (hunk 3 above)
Comment (ncknt) [reply pkg/secrets/secrets.go]: Oh yeah, it used to be in a different module. Will rename

File: pkg/halyard/halyard.go (hunk 1 above)
Comment (ncknt) [reply pkg/halyard/halyard.go]: It used to be :) Removed

**Pull request merged**